### ISSUE TITLE
Implement updated name_spec for displayForm

### ISSUE THREAD

**Issue by arcadiafalcone** [sul-dlss/dor-services-app#3215]
See #3214

**Comment by edsu** [comment]
Via @justinlittman in Slack:

In https://github.com/sul-dlss/dor-services-app/pull/3214/files a test was xited. The goal of this task is to implement the test.

**Comment by edsu** [comment]
After talking with @ndushay I decided to put this on hold until work on #3287 is merged since it was her understanding that @lwrubel was working on mapping related to parallelValue and the test I uncommented was expecting a parallelValue.

**Comment by edsu** [comment]
@lwrubel took a look and it actually looks like this might be independent of her work in subjects. So I removed the dependency on that.

**Comment by edsu** [comment]
@arcadiafalcone @justinlittman from [the spec](https://github.com/sul-dlss/dor-services-app/blob/main/spec/services/cocina/mapping/descriptive/mods/name_spec.rb#L174-L251) I gather that when a contributor `<name>` with a `<displayForm>` is present it should be split out in the Cocina as a `parallelValue`?

Is this just the case with Contributors or do we want to do this for Subjects and Titles as well? I think displayForm can appear in them too?

I think we could make the change in two places:

1. In [Cocina::FromFedora::Descriptive::NameBuilder](https://github.com/sul-dlss/dor-services-app/blob/main/app/services/cocina/from_fedora/descriptive/name_builder.rb) which is used by [Cocina::FromFedora::Descriptive::Contributor](https://github.com/sul-dlss/dor-services-app/blob/main/app/services/cocina/from_fedora/descriptive/contributor.rb) as well as [Cocina::FromFedora::Descriptive::Titles](https://github.com/sul-dlss/dor-services-app/blob/main/app/services/cocina/from_fedora/descriptive/titles.rb) and [Cocina::FromFedora::Descriptive::Subject](https://github.com/sul-dlss/dor-services-app/blob/main/app/services/cocina/from_fedora/descriptive/subject.rb).

2. In [Cocina::FromFedora::Descriptive::Contributor](https://github.com/sul-dlss/dor-services-app/blob/main/app/services/cocina/from_fedora/descriptive/contributor.rb) which would only effect Contribtors.

Do you have any thoughts/opinions about what I should try, or if I'm heading in the wrong direction? Since the spec if for names generally I was thinking that 1) is probably what was intended? But that route may entail needing to update other tests?

**Comment by arcadiafalcone** [comment]
Yes, `<name>/<namePart>` and `<name>/<displayForm>` map to `parallelValue` when both are in the same `<name>` element. `<displayForm>` in `<subject>/<name>` should already be covered in https://github.com/sul-dlss/dor-services-app/blob/59026a401130d708af654e9c922c2f7f06316a4e/spec/services/cocina/mapping/descriptive/mods/subject_name_spec.rb#L108.

**Comment by edsu** [comment]
@arcadiafalcone @justinlittman Over in #3358 I've got a draft PR where I've implemented a partially passing test. The problem is that the Cocina no longer round trips back to the expected MODS. Here is the failing test:

```
✗ bx rspec './spec/services/cocina/mapping/descriptive/mods/name_spec.rb[1:7:1:2:1]'
DEPRECATION: Rsolr.new/connect option `read_timeout` is deprecated and will be removed in Rsolr 3. `timeout` is currently a synonym, use that instead.
Run options: include {:ids=>{"./spec/services/cocina/mapping/descriptive/mods/name_spec.rb"=>["1:7:1:2:1"]}}

MODS name <--> cocina mappings
  Name with additional subelements
    behaves like MODS cocina mapping
      when mapping to MODS (from cocina)
        cocina Description maps to expected MODS (FAILED - 1)

Failures:

  1) MODS name <--> cocina mappings Name with additional subelements behaves like MODS cocina mapping when mapping to MODS (from cocina) cocina Description maps to expected MODS
     Failure/Error: expect(actual_xml).to be_equivalent_to expected_mods_ng.to_xml

       expected:
       <?xml version="1.0" encoding="UTF-8"?>
       <mods xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.loc.gov/mods/v3" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:xlink="http://www.w3.org/1999/xlink" version="3.7" xsi:schemaLocation="http://www.loc.gov/mods/v3 http://www.loc.gov/standards/mods/v3/mods-3-7.xsd">
         <name type="personal" usage="primary">
         <namePart type="termsOfAddress">Dr.</namePart>
         <namePart type="given">Terry</namePart>
         <namePart type="family">Castle</namePart>
         <namePart type="date">1953-</namePart>
         <affiliation>Stanford University</affiliation>
         <nameIdentifier type="wikidata">https://www.wikidata.org/wiki/Q7704207</nameIdentifier>
         <displayForm>Castle, Terry</displayForm>
         <description>Professor of English</description>
       </name>

       <location><url usage="primary display">https://purl.stanford.edu/zn746hz1696</url></location></mods>

       got:
       <?xml version="1.0" encoding="UTF-8"?>
       <mods xmlns="http://www.loc.gov/mods/v3" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:xlink="http://www.w3.org/1999/xlink" version="3.7" xsi:schemaLocation="http://www.loc.gov/mods/v3 http://www.loc.gov/standards/mods/v3/mods-3-7.xsd">
         <name altRepGroup="1">
           <namePart type="termsOfAddress">Dr.</namePart>
           <namePart type="given">Terry</namePart>
           <namePart type="family">Castle</namePart>
           <namePart type="date">1953-</namePart>
           <nameIdentifier type="wikidata">https://www.wikidata.org/wiki/Q7704207</nameIdentifier>
           <affiliation>Stanford University</affiliation>
           <description>Professor of English</description>
         </name>
         <name altRepGroup="1">
           <namePart>Castle, Terry</namePart>
           <nameIdentifier type="wikidata">https://www.wikidata.org/wiki/Q7704207</nameIdentifier>
           <affiliation>Stanford University</affiliation>
           <description>Professor of English</description>
         </name>
         <location>
           <url usage="primary display">https://purl.stanford.edu/zn746hz1696</url>
         </location>
       </mods>
```

Should the the `altRepGroup`s be getting flattened back into a single `<name>` here?

**Comment by edsu** [comment]
@justinlittman and I spoke this morning and he said that this really is a question for @arcadiafalcone about how we want the [parallel Cocina](https://github.com/sul-dlss/dor-services-app/blob/t3215-updated-namespec/spec/services/cocina/mapping/descriptive/mods/name_spec.rb#L194-L251) to map back to MODS.

**Comment by arcadiafalcone** [comment]
If there is a `parallelValue` in `name` where one of the values has type `display`, when converting to MODS that value should be merged into the MODS `<name>` element generated by the other `name` value in the `parallelValue`. If the `parallelValue` contains more than two name values, the `display` value should be merged into the MODS `<name>` element generated by the value immediately preceding the `display` value (i.e., smoosh it in with the one above it).

**Comment by ndushay** [comment]
@edsu which branch were you working in?

**Comment by edsu** [comment]
@ndushay the branch is https://github.com/sul-dlss/dor-services-app/tree/t3215-updated-namespec

**Comment by justinlittman** [comment]
@ndushay This can be closed, right?

**Comment by ndushay** [comment]
It's up to @arcadiafalcone, who, i think, has been frying bigger fish of late.

**Comment by arcadiafalcone** [comment]
@ndushay It was data remediation I needed to do, which is now done, so this should be ready to move forward.
